proposals/PickAPI/README.md

```diff
@@ -0,0 +1,192 @@
+
+### Visibility
+```
+uniform token pick:visibility = "inherited" (allowedTokens = ["inherited", "invisible"])
+```
+`pick:visibility` controls whether or not the object should be considered visible with respect to any picking operation. Objects `invisible` to picking should be excluded from any picking calculations (and acceleration structures like BVHs or pick buffers). Prims should only be included in the pick buffer if they are imaged, taking into account `visibility`, `purpose`, and any other viewport render settings.
+
+Because picking is often accelerated through the rendering toolkit, it's important to describe picking on its terms.
+
+If the `PickAPI` is applied to multiple sites in a hierarchy, `pick:visibility` has pruning semantics similar to the `visibility` attribute. If a prim is invisible to picking, so are its descendants.
+
+To make an object _unpickable_ but still participate in calculations as a matte occluder, users must use [retargeting](#retargeting-for-pick-occluders).
+
+`pick:visibility` may not be animated.
+
+In this current proposal, sending geometry to a pick buffer that is not actively imaged is currently out of scope. One hypothetical use case would be interactive picking of volumetrics. A user might provide a mesh to be used as the pick target for a dense volumetric cloud. This proposal doesn't provide a path for including non-imageable geometry in picking because the semantics become somewhat confusing and complicates implementations leveraging rendering pipelines.
+
+### Retargeting
+
+```
+uniform token pick:retargeting = "none" (allowedTokens = ["none", "replace"])
+rel pick:targets
+```
+These attributes control whether or not to forward picking to alternative targets. The default behavior (`none`) is to not forward picking. However, if `replace` is selected, all descendant prim selections will be replaced with `pick:targets`.
+
+If the `PickAPI` is applied to multiple sites in a hierarchy, the retargeting behavior will be defined by the nearest applied ancestor. This is notably different than the pruning semantics of `pick:visibility`. As specified, retargeting could be set to `none` in a descendant, overriding any ancestral retargeting opinions.
+
+Targets will usually include one or more prims, but may include one or more properties. If a relationship is targeted, standard relationship forwarding should be applied. If an attribute is targeted, that attribute should be selected if meaningful, otherwise the attribute's parent prim. There is no way to use the `PickAPI` to retarget selection to a relationship.
+
+#### Retargeting for Pick Occluders
+If targets is empty, no prims will be selected, and the hierarchy will act as a matte occluder. Utilities (say `UsdUIPickAPI(prim).OccludeOnly()`) can be used to explicitly set `pick:visibility` to `inherited`, `pick:retargeting` to `replace`, and `pick:targets` to an empty list.
```

Comment:
Instead of "replace to empty" to get matte occlusion, might it be clearer to have `pick:retargeting = "matteOcclusion"` or something like that?

Reply:
I considered that.  A version of this proposal had something like

```
uniform pick:retargeting = "none" (allowedTokens = ["none", "matte", "replace"])
```

It occurred to me that `replace` with empty targets would necessarily be equivalent to `matte` so I dropped the explicit option in the proposal in favor of their being one and only way to specify something. I similarly dropped a `self` mode in favor of just targeting `<.>`.  I can definitely see the reasoning for the explicit formulation, perhaps just for the `matte` case?

Reply:
That's a good concern.  Maybe it's just me, though, but I'd consider making `replace` with empty targets be an error condition, because one _could_ reasonably (I think?) assume not `matte` behavior but instead just "unpickable with passthrough" behavior?